### ISSUE TITLE
Touchables stop working on Android after using the app for a while

### ISSUE THREAD

**Issue by gabrieldonadel** [software-mansion/react-native-gesture-handler#2312]
### Description

Touchables stop working on Android after using the app for a while. I was able to reproduce this with my logcat open and when pressing on the Touchables element an error saying "Invalid ID 0x00000XXX" is thrown

![image](https://user-images.githubusercontent.com/11707729/200573812-f1dabb69-fe17-42dc-8f33-75b79d4bf080.png)

### Steps to reproduce

1. Using an Android device navigate around the app and try to press on many things at the same time
2. At some point touchable elements that use react-native-gesture-handler will stop working (this may take up to ~10 minutes to repro)

### Snack or a link to a repository

https://snack.expo.dev/@gabrieldonadel/touchables-bug

### Gesture Handler version

2.8.0

### React Native version

0.69.7

### Platforms

Android

### JavaScript runtime

Hermes

### Workflow

Expo bare workflow

### Architecture

Paper (Old Architecture)

### Build type

Release mode

### Device

Real device

### Device model

Samsung Galaxy S10 (Android 12)

### Acknowledgements

Yes

**Comment by j-piasecki** [comment]
@gabrieldonadel Hi! I believe that https://github.com/software-mansion/react-native-gesture-handler/pull/2315 fixes the problem, at least I wasn't able to reproduce it anymore with this fix applied. I would really appreciate it if you could check whether it solves the problem for you.

**Comment by gabrieldonadel** [comment]
> @gabrieldonadel Hi! I believe that #2315 fixes the problem, at least I wasn't able to reproduce it anymore with this fix applied. I would really appreciate it if you could check whether it solves the problem for you.

@j-piasecki your fix looks good to me, I've tested it on my Apps and seems to resolve the problem, thx!

**Comment by simon-abbott** [comment]
I can confirm that this is an issue in v2.7.0 through v2.8.0, and that the fix in 04f22dc8178595a3c8afb5260470649a711cee5a works (tested via [patch-package](https://www.npmjs.com/package/patch-package)). When can we expect a new release with this fix in it?

---

In case it becomes a problem again and/or anyone wants to revisit this fix further, here is a small example that reproduces the issue 100% of the time:
```tsx
<View>
  <View style={{ position: 'absolute' }}>
    <TouchableOpacity
      onPress={() => {
        console.log('Button 1 pressed');
      }}
    >
      <Text>Button 1</Text>
    </TouchableOpacity>
  </View>
  <View style={{ position: 'absolute', opacity: 0 }}>
    <TouchableOpacity
      onPress={() => {
        console.log('Button 2 pressed');
      }}
    >
      <Text>Button 2</Text>
    </TouchableOpacity>
  </View>
</View>
```
